### ISSUE TITLE
Improved error message needed: NoUri (latex builder)

### ISSUE THREAD

**Issue by shimizukawa** [sphinx-doc/sphinx#241]
When the referenced document (in toctree) is missing from filesystem, Sphinx gives a following error. While the error is valid the message is not clear - I had to look into source code and print the document name in question (which is available) to find the rootcause. Thus, the error message could be improved.

```
 File "/usr/lib/python2.4/site-packages/Sphinx-0.6.2-py2.4.egg/sphinx/cmdline.py", line 172, in main
    app.build(all_files, filenames)
  File "/usr/lib/python2.4/site-packages/Sphinx-0.6.2-py2.4.egg/sphinx/application.py", line 130, in build
    self.builder.build_update()
  File "/usr/lib/python2.4/site-packages/Sphinx-0.6.2-py2.4.egg/sphinx/builders/__init__.py", line 260, in build_update
    self.build(['__all__'], to_build)
  File "/usr/lib/python2.4/site-packages/Sphinx-0.6.2-py2.4.egg/sphinx/builders/__init__.py", line 320, in build
    self.write(docnames, list(updated_docnames), method)
  File "/usr/lib/python2.4/site-packages/Sphinx-0.6.2-py2.4.egg/sphinx/builders/latex.py", line 96, in write
    appendices=((docclass == 'manual') and
  File "/usr/lib/python2.4/site-packages/Sphinx-0.6.2-py2.4.egg/sphinx/builders/latex.py", line 153, in assemble_doctree
    self.env.resolve_references(largetree, indexfile, self)
  File "/usr/lib/python2.4/site-packages/Sphinx-0.6.2-py2.4.egg/sphinx/environment.py", line 1350, in resolve_references
    builder.app.emit('doctree-resolved', doctree, fromdocname)
  File "/usr/lib/python2.4/site-packages/Sphinx-0.6.2-py2.4.egg/sphinx/application.py", line 222, in emit
    result.append(callback(self, *args))
  File "/usr/lib/python2.4/site-packages/Sphinx-0.6.2-py2.4.egg/sphinx/ext/todo.py", line 107, in process_todo_nodes
    newnode['refuri'] = app.builder.get_relative_uri(
  File "/usr/lib/python2.4/site-packages/Sphinx-0.6.2-py2.4.egg/sphinx/builders/latex.py", line 57, in get_relative_uri
    return self.get_target_uri(to, typ)
  File "/usr/lib/python2.4/site-packages/Sphinx-0.6.2-py2.4.egg/sphinx/builders/latex.py", line 51, in get_target_uri
    raise NoUri
NoUri
```

---
- Bitbucket: https://bitbucket.org/birkenfeld/sphinx/issue/241
- Originally reported by: Juha Mustonen
- Originally created at: 2009-08-17T12:46:16.013

**Comment by shimizukawa** [comment]
_From Georg Brandl on 2009-09-03 14:11:00+00:00_

Should be fixed in <<cset 456abdefbe5167dfde70ecefd1b5ace09bb03ca9>>. Thanks for reporting!

**Comment by shimizukawa** [comment]
_From Georg Brandl on 2013-03-31 13:57:13+00:00_

Removing version: 0.6.2 (automated comment)
